### ISSUE TITLE
Compile libraries are not being populated for DependencyContext when building with an RID

### ISSUE THREAD

**Issue by joemey** [dotnet/core-setup#5447]
I'm using `DependencyContext.Default.CompileLibraries` to preload assemblies on startup.
While porting my Library to Linux and building it with a runtime specified, some of my tests started to failing.

## Steps to reproduce
```
mkdir test
cd test
dotnet new console
dotnet add package Microsoft.Extensions.DependencyModel
```

change `Program.cs` to 
```
using System;

namespace test
{
    class Program
    {
        static void Main(string[] args)
        {
            Console.WriteLine(Microsoft.Extensions.DependencyModel.DependencyContext.Default.CompileLibraries.Count);
        }
    }
}
```
## Expected  behavior

both should output the same

## Actual behavior

```
dotnet build
dotnet \bin\Debug\netcoreapp2.2\test.dll
```

this prints '40' for me

```
dotnet build -r win-x64
dotnet \bin\Debug\netcoreapp2.2\win-x64\test.dll
```
this prints `0`

Running with `donet run --runtime win-x64` does print `40` as well, but only because it seems to ignore the runtime argument and does **not** create the `win-64` subdirectory

## Environment data

Tested on Windows 10 
```
.NET Core SDK  (reflecting any global.json):
 Version:   2.2.102
 Commit:    96ff75a873

Laufzeitumgebung:
 OS Name:     Windows
 OS Version:  10.0.17763
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\2.2.102\

Host (useful for support):
  Version: 2.2.1
  Commit:  878dd11e62
```

and Ubuntu 18.10 (using `linux-x64` instead of `win-x64` as the runtime)
```
.NET Core SDK (reflecting any global.json):
 Version:   2.2.104
 Commit:    73f036d4ac

Runtime Environment:
 OS Name:     ubuntu
 OS Version:  18.10
 OS Platform: Linux
 RID:         ubuntu.18.10-x64
 Base Path:   /usr/share/dotnet/sdk/2.2.104/

Host (useful for support):
  Version: 2.2.2
  Commit:  a4fd7b2c84
```

The problem seems to be that the `.deps.json` file in the `win-x64` build contains 2 targets 
- `.NETCoreApp,Version=v2.2` which is empty
- `.NETCoreApp,Version=v2.2/win-x64` this has all libraries listed

When loading the `.deps.json` file in `DependencyContextJsonReader.ReadCore` the correct target with the RID is selected https://github.com/dotnet/core-setup/blob/8b2e20b4920cc7ff3c6181dc2b7d5ead9176e907/src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs#L73

But for the compile libraries the target without the RID is used, which is empty in this case.

https://github.com/dotnet/core-setup/blob/8b2e20b4920cc7ff3c6181dc2b7d5ead9176e907/src/managed/Microsoft.Extensions.DependencyModel/DependencyContextJsonReader.cs#L93-L102

**Comment by jeffschwMSFT** [comment]
cc @eerhardt

**Comment by eerhardt** [comment]
@joemey - can you explain why you expect the same output between running the app as framework-dependent vs running it as a stand-alone application?

The `.deps.json` files are different between these two situations (as you pointed out). So getting the exact same results isn't expected in my opinion.

You will find differences in other areas as well. For example if you print out:

`Console.WriteLine(DependencyContext.Default.RuntimeGraph.Count);`

You will also get `0` for the stand-alone case, and non-zero for a framework-dependent app.

The real surprise to me is that `CompileLibraries` is not empty for framework-dependent app. I thought this would only be filled out when `PreserveCompilationContext=true` in the `.csproj`, but apparently that isn't the case.

So, I'd like to understand:

1. Why you think these two different scenarios should behave the same?
2. Which would you expect it to be: both `0` or both `44`?

Do you have a higher scenario here you'd like to enable?

**Comment by joemey** [comment]
I think I was expecting the same output because it "works" when i'm not specifing a runtime without setting `PreserveCompilationContext=true`.

My scenario is that I have a set of libraries (managed and unmanged) and I want to preload/setup some data on startup. While adding linux support some of my tests started to fail because of this.

Maybe I'm just using the wrong property and I should use `RuntimeLibraries` instead.

**Comment by eerhardt** [comment]
Yes, for that scenario RuntimeLibraries is exactly what you should be using. ComplileLibraries is for when you want to know which libraries were used to compile the application. This is useful when you want to compile additional code at runtime - which is what ASP.NET Razor pages will do.

RuntimeLibraries are the libraries that will be loaded and executed when the app runs.

Note that there are some NuGet packages that have different “ref” vs “lib” assemblies. The “ref” libraries are what are added to CompileLibraries. And sometimes these libraries only contain the public method signatures without any implementation in them. So attempting to load these libraries at runtime will not work.
